### PR TITLE
wifi_manager: add wifi manager error code

### PR DESCRIPTION
**Pull request by pillip8282**
To distinguish post fail and driver fail, add wifi_manager_post_fail
error type.

### CODE REVIEW

**Review by jeongchanKim — COMMENTED** [review]
Please check the indentation.

File: framework/src/wifi_manager/wifi_manager_utils.h [hunk 1]
```diff
@@ -72,6 +72,18 @@ int net_task_create(FAR const char *name, int priority, int stack_size,
 		}												\
 	} while (0)
 
+/* errno EINVAL shouldn't be happened
+ * if it were happened then system could be corrupted */
```
Comment (jeongchanKim) [inline framework/src/wifi_manager/wifi_manager_utils.h:76]: `were` -> `was`

Comment (abhishek-samsung) [inline framework/src/wifi_manager/wifi_manager_utils.h:76]: Could you please change "errno EINVAL shouldn't be happened * if it were happened then system could be corrupted" to "errno EINVAL shouldn't happen * if it happens then the system could be corrupted"

File: framework/include/wifi_manager/wifi_manager.h [hunk 2]
```diff
@@ -90,6 +90,7 @@ typedef enum {
 	WIFI_MANAGER_DISCONNECT,
 	WIFI_MANAGER_RECONNECT, //AUTOCONNECT
 	WIFI_MANAGER_NO_API,
+  WIFI_MANAGER_POST_FAIL,
```
Comment (kishore-sn) [inline framework/include/wifi_manager/wifi_manager.h:93]: Please check alignment for this

File: framework/src/wifi_manager/wifi_manager_msghandler.c [hunk 3]
```diff
@@ -97,13 +96,8 @@ int wifimgr_post_message(wifimgr_msg_s *msg)
 		return -1;
 	}
 
-	res = sem_wait(hmsg.signal);
-	if (res == -1) {
-		NET_LOGE(TAG, "wait msg signal fail %d\n", errno);
-		return -2;
-	}
+  WIFIMGR_WAIT_SIG(hmsg.signal);
```
Comment (kishore-sn) [inline framework/src/wifi_manager/wifi_manager_msghandler.c:99]: Please check alignment.

File: framework/src/wifi_manager/wifi_manager_utils.h (hunk 1 above)
Comment (pillip8282) [reply framework/src/wifi_manager/wifi_manager_utils.h:76]: Thanks I'll modify

File: framework/include/wifi_manager/wifi_manager.h (hunk 2 above)
Comment (pillip8282) [reply framework/include/wifi_manager/wifi_manager.h:93]: Thanks I'll modify

**Review by an4967 — APPROVED** [review]
LGTM

**Pull request merged**